### PR TITLE
fixed react unmount component error

### PR DESCRIPTION
**Pull request by zori7**

### CODE REVIEW

File: docusaurus/src/components/CodeFromFile/CodeFromFile.tsx [hunk 1]
```diff
@@ -37,12 +37,22 @@ const CodeFromFile = ({
   }, [siteConfig, filePath]);
 
   useEffect(() => {
-    fetch(url)
+    console.log('UseEffect');
```
Comment (bmeeder22) [inline docusaurus/src/components/CodeFromFile/CodeFromFile.tsx:40]: Remove this

File: docusaurus/src/components/CodeFromFile/CodeFromFile.tsx [hunk 2]
```diff
@@ -37,12 +37,22 @@ const CodeFromFile = ({
...
+    const controller = new AbortController();
+    fetch(url, {signal: controller.signal})
       .then(response => response.text())
       .then(data => setCode(data))
+      .catch(error => {
+        if (error.code === 20) return;
+
+        return Promise.reject(error);
```
Comment (bmeeder22) [inline docusaurus/src/components/CodeFromFile/CodeFromFile.tsx:48]: We're not using this rejected promise anywhere

File: docusaurus/src/components/CodeFromFile/CodeFromFile.tsx [hunk 3]
```diff
@@ -37,12 +37,22 @@ const CodeFromFile = ({
...
 
   useEffect(() => {
-    fetch(url)
+    console.log('UseEffect');
+    const controller = new AbortController();
+    fetch(url, {signal: controller.signal})
       .then(response => response.text())
       .then(data => setCode(data))
+      .catch(error => {
+        if (error.code === 20) return;
```
Comment (bmeeder22) [inline docusaurus/src/components/CodeFromFile/CodeFromFile.tsx:46]: Would this not be 200?

**Review by bmeeder22 — COMMENTED** [review]
Just some questions

File: docusaurus/src/components/CodeFromFile/CodeFromFile.tsx (hunk 3 above)
Comment (zori7) [reply docusaurus/src/components/CodeFromFile/CodeFromFile.tsx:46]: Sorry for confusion, forgot to init a const for the error code, it's aborted request code

File: docusaurus/src/components/CodeFromFile/CodeFromFile.tsx (hunk 2 above)
Comment (bmeeder22) [reply docusaurus/src/components/CodeFromFile/CodeFromFile.tsx:48]: You can just do .catch(()=> {}) 🙂

**Review by bmeeder22 — APPROVED** [review]

**Pull request merged**